plugins/destination/postgresql/client/migrate.go

```diff
@@ -211,21 +213,42 @@ func (c *Client) migrateToCQID(ctx context.Context, table *schema.Table, _ schem
 		c.logger.Error().Err(err).Str("table", tableName).Msg("Failed to drop primary key")
 		return err
 	}
+	// Create new Primary Key with CQID
+	_, err = tx.Exec(ctx, "ALTER TABLE "+sanitizedTableName+" ADD CONSTRAINT "+sanitizedPKName+" PRIMARY KEY ("+pgx.Identifier{schema.CqIDColumn.Name}.Sanitize()+")")
+	if err != nil {
+		c.logger.Error().Err(err).Str("table", tableName).Msg("Failed to create new primary key on _cq_id")
+		return err
+	}
 
-	for _, colName := range table.PrimaryKeyComponents() {
+	// CockroachDB doesn't support dropping NOT NULL constraints in the same transaction as the the primary key is changed
+	// So we have to alter the PK in one transaction and then drop the old NOT NULL constraints in another transaction
+	if c.pgType == pgTypeCockroachDB {
+		if err == nil {
+			err = tx.Commit(ctx)
+			if err != nil {
+				c.logger.Error().Err(err).Msg("failed to commit transaction")
+			}
+		}
+		if err != nil {
+			if rollbackErr := tx.Rollback(ctx); rollbackErr != nil {
+				c.logger.Error().Err(rollbackErr).Str("table", tableName).Msg("Failed to rollback transaction")
+			}
+		}
+		tx, err = conn.BeginTx(ctx, pgx.TxOptions{
+			IsoLevel: pgx.Serializable,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to begin transaction: %w", err)
+		}
+	}
+	for _, colName := range c.pgTablesToPKConstraints[tableName].columns {
```

Comment:
```suggestion
	entry, _ := c.pgTablesToPKConstraints[tableName]
	if entry == nil {
		entry = new(pkConstraintDetails)
	}
	for _, colName := range entry.columns {
```

Reply:
All tables are guaranteed to be in the map at this point

Reply:
Done in 1ee5ff6ca48caa7ae4bf950f406f026c85f54f70